src/lib/registration/ui/not_installed_products_dialog.rb

```diff
@@ -0,0 +1,161 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2017 SUSE LLC
+#
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE.
+#
+# To contact SUSE about this file by physical or electronic mail, you may find
+# current contact information at www.suse.com.
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "registration/sw_mgmt"
+require "registration/helpers"
+
+module Registration
+  module UI
+    # this class displays and runs the dialog which check all the installed
+    # but not registered products warning the user about it and allowing him
+    # to take some actions over them.
+    class NotInstalledProductsDialog
+      include Yast::Logger
+      include Yast::I18n
+      include Yast::UIShortcuts
+      include Yast
+
+      attr_accessor :registration, :registration_ui
+
+      def self.run
+        dialog = NotInstalledProductsDialog.new
+        dialog.run
+      end
+
+      def initialize
+        textdomain "registration"
+
+        self.registration = Registration.new(UrlHelpers.registration_url)
+        self.registration_ui = RegistrationUI.new(registration)
+      end
+
+      def run
+        Yast::Popup.Feedback(RegistrationUI::CONTACTING_MESSAGE, _("Checking product conflicts")) do
+          return :next if !registered_not_installed_addons?
+        end
+
+        Yast::UI.OpenDialog(Opt(:decorated), content)
+
+        begin
+          handle_dialog
+        ensure
+          Yast::UI.CloseDialog
+        end
+      end
+
+      private
+
+      def content
+        VBox(
+          MinWidth(80,
+            MinHeight(15,
+              VWeight(25, RichText(Id(:summary), not_installed_addons_summary))
+            )
+          ),
+          ButtonBox(
+            PushButton(Id(:cancel), Opt(:key_F9, :cancelButton), Yast::Label.AbortButton),
+            # FIXME: Maybe we could remove this option and just warn the user
+            PushButton(Id(:install), _("Ins&tall addons")),
+            PushButton(Id(:sync), _("&Synchronize")),
```

Comment:
The help label says *Synchronize your products if you want to deactivate them at your registration server* so I'd use *Deactivate* button label here and describe what does it mean in the help label.